src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -114,6 +114,15 @@ class ListBuffer[A]
     this
   }
 
+  // Overridden for performance (to avoid virtual call to addOne
+  override final def addAll(xs: IterableOnce[A]): this.type = {
```

Comment:
we should check for empty to avoid the iterator allocation

Reply:
Implementations of `.iterator` do that internally and return a shared instance of an empty iterator ([example](https://github.com/scala/scala/blob/f293db457208e1c009333120296935605d20446e/src/library/scala/collection/immutable/Vector.scala#L107-L115)).

Reply:
doh - forgot that